graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/TopSecondDegreeByCountTweetRecsGenerator.java

```diff
@@ -109,6 +105,33 @@ private static boolean isLessThantMinUserSocialProofSize(
     return !keep;
   }
 
+  private static boolean isLessThanMinUserSocialProofSizeCombined(
+    SmallArrayBasedLongToDoubleMap[] socialProofs,
+    int minUserSocialProofSize,
+    Set<byte[]> socialProofTypeUnions
+  ) {
+    boolean keep = false;
+
+    for (byte[] socialProofTypeUnion: socialProofTypeUnions) {
+      Set<Long> uniqueNodes = new HashSet<Long>();
```

Comment:
it is better to avoid autoboxing and unboxing whenever we can. Prefer LongOpenHashSet which stores primitive long directly instead of Long object.